File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr) {
    
    if(arr.size() == arr[0].size()) return arr;
    int row = arr.size();
    int col = arr[0].size();
    if(row > col) col = row;
    else row = col;
    vector<vector<int>> answer(row, vector<int>(col));
    for(int i=0;i<row;i++){
        for(int j=0;j<col;j++){
            if(i >= arr.size())
                answer[i][j] = 0;
            else if(j >= arr[i].size())
                answer[i][j] = 0;
            else
                answer[i][j] = arr[i][j];
        }
    }
    return answer;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    if len(arr) == len(arr[0]):
        return arr
    row, col = len(arr), len(arr[0]);
    if row > col:
        col = row
    else:
        row = col
    answer = [ [0 for _ in range(col)] for _ in range(row)]
    for i in range(row):
        for j in range(col):
            if i >= len(arr):
                continue
            elif j >= len(arr[0]):
                continue
            else:
                answer[i][j] = arr[i][j]
    return answer